runtime/permissions/lib.rs

```diff
@@ -2317,6 +2342,36 @@ pub fn is_standalone() -> bool {
   IS_STANDALONE.is_raised()
 }
 
+static WILDCARD_PERMISSIONS: Lazy<RwLock<Vec<String>>> =
+  Lazy::new(|| RwLock::new(Vec::new()));
+
+pub fn add_wildcard_permission(permission: &str) -> Result<(), String> {
+  let mut permissions = WILDCARD_PERMISSIONS.write().map_err(|e| {
+    error!("Failed to acquire write lock: {}", e);
+    "Failed to add wildcard permission".to_string()
+  })?;
+  if !permissions.contains(&permission.to_string()) {
+    permissions.push(permission.to_string());
+  }
+  Ok(())
+}
+
+pub fn get_wildcard_permissions() -> Vec<String> {
+  let permissions = WILDCARD_PERMISSIONS
+    .read()
+    .expect("Failed to acquire read lock");
+  permissions.clone()
+}
+
+pub fn clear_wildcard_permissions() -> Result<(), String> {
+  let mut permissions = WILDCARD_PERMISSIONS.write().map_err(|e| {
+    error!("Failed to acquire write lock: {}", e);
+    "Failed to clear wildcard permissions".to_string()
+  })?;
+  permissions.clear();
+  Ok(())
+}
```

Comment:
Can you explain the purpose of this code? What's the reason for having a static with a lock just to match permissions?

Reply:
@bartlomieju 

The code uses a RwLock to handle wildcard permissions dynamically. When a wildcard is passed (e.g., `--allow-env=MYAPP_*`), the logic captures and stores these patterns. Before checking an environment variable's permission, the stored patterns are matched to grant access. The lock ensures thread-safe reads/writes while allowing concurrent reads. **This dynamic approach helps manage permissions for env variables set by the user (via `Deno.env.set`), even if they weren’t present at the script’s start. It addresses the issue where wildcard permissions don’t automatically allow newly set env variables like F`OOBAR_X.`**

Reply:
That is not correct. You want to solve the same way `--allow-net` permissions are stored - you store an enum of either a concrete value (`--allow-env=NODE_DEBUG`) or a wildcard match (`--allow-env=AWS_*`, `--allow-env=*_DENO`). Then when the permission is checked you either perform an exact match on "concrete value" enum variant, or a wildcard match on the other variant. The `RwLock` is not needed in here. 

So you need to update `EnvDescriptor` to something like:
```
#[derive(Clone, Eq, PartialEq, Hash, Debug)]
pub enum EnvDescriptor {
  Concrete(EnvVarName),
  Wildcard(...)
}
```
and then `impl UnaryPermission<EnvDescriptor>` to apply that logic.

Reply:
@bartlomieju ,
its cant be Enum maybe have many suffixes like this  AWS_*,FOOBAR_* , many prefix *_DENO, *_EXALT,
how theses random different name can be stores in Enum

Reply:
You store one element in a single enum - then multiple enums are stored here: https://github.com/denoland/deno/blob/b01578ae1fc1da65ac38daec31a23c9ef652aa74/runtime/permissions/lib.rs#L338-L340

Reply:
> You store one element in a single enum - then multiple enums are stored here:
> 
> https://github.com/denoland/deno/blob/b01578ae1fc1da65ac38daec31a23c9ef652aa74/runtime/permissions/lib.rs#L338-L340

@bartlomieju 
done check latest commit